src/Ushahidi/Modules/V5/Http/Controllers/PostController.php

```diff
@@ -354,15 +354,26 @@ public function delete(int $id, Request $request)
 
     public function stats(Request $request)
     {
-        $stats = $this->queryBus->handle(PostsStatsQuery::FromRequest($request));
+        $surveys_with_private_location  = $this->queryBus->handle(new GetSurveyIdsWithPrivateLocationQuery());
```

Comment:
This query should be moved into the if clause and only be executed if the user cannot see posts with private location.